### ISSUE TITLE
difference between iterblocks and nchunks

### ISSUE THREAD

**Issue by simonm3** [Blosc/bcolz#388]
From the code below there are 62 blocks but nchunks is 61. Why is that?

```
import bcolz
a=bcolz.carray(np.array([1,2,3,4]*999999))
print(len(list(bcolz.iterblocks(a))), a.nchunks)
```

**Comment by FrancescAlted** [comment]
Yes, that's intended.  `a.nchunks` is the number of chunks that are complete, whereas `iterblocks` iterates over the `remainder` block.

**Comment by simonm3** [comment]
Thanks. Would be useful to add something to the docs on the difference
between blocks and chunks. I searched for it but didn't find anything.
